handler_test.go

```diff
@@ -0,0 +1,1982 @@
+				// Wait for it to be informer to get the events.
+				err := wait.PollUntilContextTimeout(context.Background(), 100*time.Millisecond, 1*time.Minute, true, func(ctx context.Context) (done bool, err error) {
+
+					// Get the current DNSNameResolver object.
+					resolverObj, err = lister.DNSNameResolvers("dns").Get(dnsNameResolver.Name)
+					if err != nil {
+						t.Fatalf("error retrieving dns name resolver add: %v", err)
+					}
+
+					matched := false
+
+					// Check if the number of resolved names in the objects's status and in the expected status is equal to zero.
+					if len(dnstc.expectedStatuses[index].ResolvedNames) == 0 && len(resolverObj.Status.ResolvedNames) == 0 {
+						matched = true
+					}
+
+					// Check if the number of resolved names in the status matches that of the expected status.
+					if len(dnstc.expectedStatuses[index].ResolvedNames) == len(resolverObj.Status.ResolvedNames) {
+
+						// Iterate through each resolved name in the object's status and check if it matches that of the expected status.
+						for index, expectedResolvedName := range dnstc.expectedStatuses[index].ResolvedNames {
+							matched = false
+							currentResolvedName := resolverObj.Status.ResolvedNames[index]
+							if currentResolvedName.DNSName == expectedResolvedName.DNSName {
+								cmpOpts := []cmp.Option{
+									cmpopts.IgnoreFields(metav1.Condition{}, "ObservedGeneration", "LastTransitionTime"),
+									cmpopts.IgnoreFields(ocpnetworkapiv1alpha1.DNSNameResolverResolvedAddress{}, "LastLookupTime"),
+									cmpopts.SortSlices(func(elem1, elem2 ocpnetworkapiv1alpha1.DNSNameResolverResolvedAddress) bool {
+										return elem1.IP > elem2.IP
+									}),
+								}
+								if cmp.Diff(currentResolvedName, expectedResolvedName, cmpOpts...) == "" {
+									matched = true
+								}
+							}
+							if !matched {
+								break
+							}
+						}
+					}
+					if !matched {
+						return false, nil
+					}
+
+					return true, nil
+				})
+				// If the object's status did not match the expected status then fail the test.
+				if err != nil {
+					t.Logf("Current Status: %v", resolverObj.Status)
+					t.Logf("Expected Status: %v", dnstc.expectedStatuses[index])
+					t.Fatalf("dns name resolver object's status did not match the expected status: DNS name: %s", resolverObj.Spec.Name)
+				}
+
+				// Delete the DNSNameResolver object.
+				err = fakeNetworkClient.NetworkV1alpha1().DNSNameResolvers("dns").Delete(context.TODO(), dnsNameResolver.Name, metav1.DeleteOptions{})
+				if err != nil {
+					t.Fatalf("error deleting dns name resolver: %v", err)
+				}
+
+				// Wait for it to be informer to get the event.
+				err = wait.PollUntilContextTimeout(context.Background(), 100*time.Millisecond, 1*time.Minute, true, func(ctx context.Context) (done bool, err error) {
+					_, err = lister.DNSNameResolvers("dns").Get(dnsNameResolver.Name)
+					if err == nil || !kerrors.IsNotFound(err) {
+						return false, nil
+					}
```

Comment:
Shouldn't this be rather something like:
```suggestion
					if err != nil || !kerrors.IsNotFound(err) {
						return false, err
					}
```

Reply:
Here we  are checking if the object is deleted. If there are no error then the object is fetched successfully. We should return true only if the object is "NotFound", otherwise return an error on context timeout.

Reply:
Ah, ok, I think the comment before the poll function misled me:

> Wait for it to be informer to get the event.

Should be be rather:
```
// Wait for informer to get the delete event.
```